src/Savegame/AreaSystem.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+/*
+ * Copyright 2010-2024 OpenXcom Developers.
+ *
+ * This file is part of OpenXcom.
+ *
+ * OpenXcom is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * OpenXcom is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenXcom.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <entt/entt.hpp>
+
+namespace OpenXcom
+{
+
+/**
+ * @brief defines a type that has a longitude and latitude
+ */
+template<typename T>
+concept LocatableConcept = requires(T locatable) {
```

Comment:
👍

Reply:
Yeah currently all these types descend from a `Target` type. Potentially this can be done away with.